test/js/bun/shell/exec.test.ts

```diff
@@ -0,0 +1,25 @@
+import * as t from "bun:test";
+import { TestBuilder } from "./test_builder";
+import { bunEnv } from "harness";
+
+const BUN = process.argv0;
+
+t.describe("bun exec", () => {
+  TestBuilder.command`${BUN} exec ${"echo hi!"}`.env(bunEnv).stdout("hi!\n").runAsTest("it works");
```

Comment:
Where is the test checking that a script exiting with a non-zero exit code is propagated to the parent process? We should also have a test which prints a large amount of output to stdout or stderr.